automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala

```diff
@@ -63,6 +67,50 @@ class SamApiSpec extends FreeSpec with Matchers {
       removePet(userStatus.userInfo)
       findPetInGoogle(userStatus.userInfo) shouldBe None
     }
+
+    def cleanupUser(subjectId: WorkbenchUserServiceAccountSubjectId): Unit = {
+      implicit val token: AuthToken = UserPool.chooseAdmin.makeAuthToken()
+      if (Sam.admin.doesUserExist(subjectId.value).getOrElse(false)) {
+        Sam.admin.deleteUser(subjectId.value)
+      }
+      Thurloe.keyValuePairs.deleteAll(subjectId.value)
+    }
+
+    def registerSaAsNewUser(email: WorkbenchUserServiceAccountEmail)(implicit authToken: AuthToken): Unit = {
+      val newUserProfile = Orchestration.profile.BasicProfile (
+        firstName = "Generic",
+        lastName = "Testerson",
+        title = "User",
+        contactEmail = Option(email.value),
+        institute = "Broad",
+        institutionalProgram = "DSP",
+        programLocationCity = "Cambridge",
+        programLocationState = "MA",
+        programLocationCountry = "USA",
+        pi = "Albus Dumbledore",
+        nonProfitStatus = "true"
+      )
+      Orchestration.profile.registerUser(newUserProfile)
+    }
+
+    "should not treat non-pet service accounts as pets" in {
+      val saEmail = WorkbenchUserServiceAccountEmail(Config.GCS.qaEmail)
+      val sa = findSaInGoogle(saEmail.toAccountName).get
+
+      // ensure clean state: SA's user not registered
+      cleanupUser(sa.subjectId)
```

Comment:
Question - will "cleaning up" the QA service account affect anything else? Is it being used elsewhere?

Reply:
I wondered about that -- but "clean up" seems to just clean up in Sam, NOT delete the SA in Google (which would be bad). @jmthibault79 confirm?

Reply:
This removes the SA's *associated Firecloud user* which did not exist before I made this test